src/Bicep.LangServer.IntegrationTests/ParamsCompletionTests.cs

```diff
@@ -216,52 +204,55 @@ public async Task Request_for_parameter_identifier_completions_should_return_cor
 new CompletionItemKind[] { CompletionItemKind.EnumMember, CompletionItemKind.EnumMember }
 )
 ]
-        public async Task Request_for_parameter_allowed_value_completions_should_return_correct_value(string paramText, string bicepText, string[] completionLables, CompletionItemKind[] completionItemKinds)
+        public async Task Request_for_parameter_allowed_value_completions_should_return_correct_value(string paramTextWithCursor, string bicepText, string[] expectedLabels, CompletionItemKind[] expectedKinds)
         {
-            var (paramFileTextNoCursor, cursor) = ParserHelper.GetFileWithSingleCursor(paramText);
-
-            var paramUri = InMemoryFileResolver.GetFileUri("/path/to/param.bicepparam");
-            var paramFile = SourceFileFactory.CreateBicepFile(paramUri, paramFileTextNoCursor);
-
             var fileTextsByUri = new Dictionary<Uri, string>
             {
-                [paramUri] = paramFileTextNoCursor,
                 [InMemoryFileResolver.GetFileUri("/path/to/main.bicep")] = bicepText
             };
 
-            using var helper = await LanguageServerHelper.StartServerWithText(
-                TestContext,
-                fileTextsByUri,
-                paramUri,
-                services => services
-                    .WithNamespaceProvider(BuiltInTestTypes.Create())
-                    .WithFeatureOverrides(new(TestContext, ParamsFilesEnabled: true)));
-
-            var file = new FileRequestHelper(helper.Client, paramFile);
-
-            var completions = await file.RequestCompletion(cursor);
+            var completions = await RunCompletionScenario(paramTextWithCursor, fileTextsByUri.ToImmutableDictionary());
 
             var expectedValueIndex = 0;
             foreach (var completion in completions)
             {
-                completion.Label.Should().Be(completionLables[expectedValueIndex]);
-                completion.Kind.Should().Be(completionItemKinds[expectedValueIndex]);
+                completion.Label.Should().Be(expectedLabels[expectedValueIndex]);
+                completion.Kind.Should().Be(expectedKinds[expectedValueIndex]);
                 expectedValueIndex += 1;
             }
         }
 
+        [DataRow(@"using './main.bicep'
+param myInt = |",@"param myInt int", new string[0])]
+        [DataRow(@"using './main.bicep'
+param myBool = |", @"param myBool bool", new[] {"false", "true"})]
+        [DataRow(@"using './main.bicep'
+param myBool =|", @"param myBool bool", new[] { "false", "true" })]
+        [DataRow(@"using './main.bicep'
+param myArray = |", @"param myArray array", new[] { "[]" })]
+        [DataRow(@"using './main.bicep'
+param myArray =|", @"param myArray array", new[] { "[]" })]
+        [DataRow(@"using './main.bicep'
+param myObj = |", @"param myObj object", new[] { "{}" })]
+        [DataRow(@"using './main.bicep'
+param myObj =|", @"param myObj object", new[] { "{}" })]
+        [DataTestMethod]
```

Comment:
Should we add a test for allowed values?

Reply:
The interns added one, so I left it as-is.

Reply:
Sounds good.